### PR TITLE
Fix #171

### LINKED ISSUE #171: SSR/Cache bug when opening user profiles
**Describe the bug**
loading a users profile always contains all data from all users on the first fresh request. 
hitting f5 fixes the issue until you get to the profile again

**To Reproduce**
1. Authenticate '
2. Open your own profile
3. see all posts from all users loaded (count and recent posts)

**Expected behavior**
only see posts related to this particular user

### PR DESCRIPTION
**Pull request by roschaefer**
I still don't understand why this does not happen on the server...

### CODE REVIEW

File: server/hooks/exclude-blacklisted.js [hunk 1]
```diff
@@ -13,8 +13,11 @@ module.exports = function excludeBlacklisted() {
       if (usersettings.total <= 0){
         return hook;
       }
-      const blacklist = usersettings.data[0].blacklist;
-      hook.params.query.userId = {$nin: blacklist};
```
Comment (roschaefer) [inline server/hooks/exclude-blacklisted.js]: the `query.userId` got overwritten

**Comment by roschaefer** [comment]
for some strange reason `hook.params.query.userId` gets set to `{ '$ne': '5bb78a5c1afa23002c128dfa' }`. Where does that come from? I cannot see any `query.userId` in the code base?

File: server/hooks/exclude-blacklisted.js [hunk 2]
```diff
@@ -13,8 +13,12 @@ module.exports = function excludeBlacklisted() {
...
+      const { blacklist } = usersettings.data[0];
```
Comment (mimicc83) [inline server/hooks/exclude-blacklisted.js]: is there any case usersettings.data could not be set or null?

**Comment by roschaefer** [comment]
@appinteractive it's `params.query.userId` not `params.userId`

**Comment by appinteractive** [comment]
@roschaefer yeah you are right! But search for `userId:` in the webapp, there are planty of places where it is used. 

When you are changing it, you would have to not overwrite it but merge it like so:
```
params.query.userId = { ...{ $nin: blacklist }, ...params.query.userId }
```

**Comment by appinteractive** [comment]
Okay now I see it, just saw the removed part, not the new one somehow.

**Comment by appinteractive** [comment]
Still, hotfix for master or develop only?

**Pull request merged**